std/range/package.d

```diff
@@ -4125,32 +4125,60 @@ pure unittest
     Generate lockstep's opApply function as a mixin string.
     If withIndex is true prepend a size_t index to the delegate.
 */
-private string lockstepMixin(Ranges...)(bool withIndex)
+private string lockstepMixin(Ranges...)(bool withIndex, bool reverse)
 {
     import std.format : format;
 
     string[] params;
     string[] emptyChecks;
     string[] dgArgs;
     string[] popFronts;
+    string indexDef;
+    string indexInc;
 
     if (withIndex)
     {
         params ~= "size_t";
         dgArgs ~= "index";
+        if (reverse)
+        {
+            indexDef = q{
+                size_t index = ranges[0].length-1;
+                enforce(_stoppingPolicy == StoppingPolicy.requireSameLength, "lockstep can only be used with foreach_reverse when stoppingPolicy == requireSameLength");
```

Comment:
> This is the current behavior:
>
> ```
> foreach_reverse (e1, e2; lockstep([1,2,3], [4,5])) {
> assert(e1 == e2 - 2);
> }
> ```

I guess that's why I got confused.
@DmitryOlshansky did you mean with "Also would prefer current behaviour." disallowing `shortest"?
I think I will just stop annoying you - I am wasting your time on such a rare edge case that can also be solved easier e.g. with `retro` - s.t. it took a longer time for its lack even to be discovered.

> break symmetry with regular foreach_reverse (indexes would be wrong)

```
foreach_reverse (i, e1, e2; lockstep([1,2,3], [4,5])) {
  // (2,3,5), (1, 4, 2)
}
```

> For non-indexed foreach_reverse, shortest is already supported and behaves the way you want:

Test please ;-)